components/eam/src/dynamics/se/stepon.F90

```diff
@@ -516,15 +523,30 @@ subroutine stepon_run3(dtime, cam_out, phys_state, dyn_in, dyn_out)
        call scm_setinitial(elem)
        if (masterproc) call readiopdata(iop_update_phase1,hyam,hybm)
        call scm_broadcast()
-       call scm_setfield(elem,iop_update_phase1)
-     endif   
+       if (.not. dp_crm) call scm_setfield(elem,iop_update_phase1)
+     endif  
 
-   endif   
+   endif
 
    call t_barrierf('sync_dyn_run', mpicom)
    call t_startf ('dyn_run')
    call dyn_run(dyn_out,rc)
-   call t_stopf  ('dyn_run')  
+   call t_stopf  ('dyn_run')
+
+   if (dp_crm) then
+
+     do ie=1,nelemd
+       do j=1,np
+         do i=1,np
+           out_gridx(i+(j-1)*np) = dyn_in%elem(ie)%spherep(i,j)%lat
+           out_gridy(i+(j-1)*np) = dyn_in%elem(ie)%spherep(i,j)%lon
```

Comment:
Is this unwrapping to a 1d array needed?
Just looking at DYN_PS output in stepon, that does not need 1D array:
`      call outfld('DYN_PS'    ,p_i(:,:,nlevp),npsq,ie)`